### PR TITLE
fix(dockerpull): fix logic for image pull for dockerfile apps

### PR DESCRIPTION
**Pull request by smothiki**
to create service we have to know port . 
This fixes logic for pulling deis pull app images and dockerfile type app images for inspecting Port

### CODE REVIEW

**Comment by smothiki** [comment]
also fixes https://github.com/deis/workflow/issues/312

**Comment by helgi** [comment]
Problem here is that we are falling back on what the user defines as `PORT` and hoping that docker is right. Probably is the best thing but just wanted to point out that now what the user defines is a fallback

**Comment by smothiki** [comment]
in that case we should give user to configure health check ports( I think this feature is already there) and use conditions that use configured port first and if not set then fall back to ports from spec and docker image. Let me know if health checks are configurable using ```deis config``` I'll add logic accordingly.

**Comment by helgi** [comment]
They should be, the `PORT` logic was to get past certain limitations before, less worried about it now.

**Comment by smothiki** [comment]
Also creates service first and then RC. helps in probing service for target port

File: rootfs/scheduler/__init__.py [hunk 1]
```diff
@@ -770,8 +772,9 @@ def _create_service(self, name, app_name, app_type, data={}, **kwargs):
         image = kwargs.get('image')
         try:
             image = self.registry + '/' + image
+            repo = image.split(":")
             # image already includes the tag, so we split it out here
-            docker_cli.pull(image.rsplit(':')[0], image.rsplit(':')[1])
+            docker_cli.pull(repo[0]+":"+repo[1], tag=repo[2], insecure_registry=True)
```
Comment (helgi) [inline rootfs/scheduler/__init__.py]: Why do we need `insecure_registry` here? Because we are generally doing `insecure` in deis right now?

**Comment by helgi** [comment]
Nice, much cleaner overall logic

**Comment by smothiki** [comment]
```
spec:
  containers:
  - env:
    - name: DEIS_APP
      value: loving-jerrycan
    - name: DEIS_RELEASE
      value: v2
    image: 10.0.96.168:5000/loving-jerrycan:git-be3be457
    imagePullPolicy: IfNotPresent
    livenessProbe:
      httpGet:
        path: /
        port: 80
        scheme: HTTP
      initialDelaySeconds: 10
      timeoutSeconds: 1
    name: loving-jerrycan-cmd
    readinessProbe:
      httpGet:
        path: /
        port: 80
        scheme: HTTP
      initialDelaySeconds: 10
      timeoutSeconds: 1
    resources: {}
```

File: rootfs/scheduler/__init__.py (hunk 1 above)
Comment (bacongobbler) [inline rootfs/scheduler/__init__.py]: The internal registry is insecure and the docker engine demands that you explicitly set this flag when dealing with these, however I'm not sure why we need to set it in the CLI if the engine already has such information.

Comment (carmstrong) [inline rootfs/scheduler/__init__.py]: :+1: I don't like the idea of having this logic live in the component unless it's absolutely necessary

Comment (smothiki) [inline rootfs/scheduler/__init__.py]: I copied this line from publish_release thought a best practice 
https://github.com/deis/workflow/blob/master/rootfs/registry/dockerclient.py#L54

Comment (smothiki) [inline rootfs/scheduler/__init__.py]: @bacongobbler do you want me to remove insecure registry flag?

Comment (bacongobbler) [inline rootfs/scheduler/__init__.py]: yes please. we should remove that flag elsewhere in a followup :)

**Comment by arschles** [comment]
@smothiki what is the manifest that you posted ^^?

Anyway, LGTM

**Comment by smothiki** [comment]
Its the APP manifest but not complete one  Just the health check part

**Pull request merged**